Framework/API/src/SpectrumInfo.cpp

```diff
@@ -212,8 +210,8 @@ SpectrumInfo::getDetectorVector(const size_t index) const {
 }
 
 std::vector<size_t> SpectrumInfo::getDetectorIndices(const size_t index) const {
-  const auto &specDef =
-      m_experimentInfo.internalSpectrumInfo().spectrumDefinition(index);
+  m_experimentInfo.updateSpectrumDefinitionIfNecessary(index);
```

Comment:
I found it a little odd to see these calls from a const getter. Presumably this is part of the lazy initialisation?